rust/lance/src/dataset/write/merge_insert/exec/write.rs

```diff
@@ -33,23 +34,30 @@ use crate::{
     },
     Dataset, Result,
 };
-use lance_core::ROW_ADDR;
-use lance_table::format::Fragment;
+use lance_core::{Error, ROW_ADDR};
+use lance_table::format::{Fragment, RowIdMeta};
+use snafu::location;
 use std::collections::BTreeMap;
 
 /// Shared state for merge insert operations to simplify lock management
 struct MergeState {
     /// Row addresses that need to be deleted, due to a row update or delete action
     delete_row_addrs: RoaringTreemap,
+    /// Shared collection to capture row ids that need to be updated
+    updating_row_ids: Arc<Mutex<CapturedRowIds>>,
     /// Merge operation metrics
     metrics: MergeInsertMetrics,
+    /// Whether to enable stable row ids
+    enable_stable_row_ids: bool,
```

Comment:
Merge insert doesn't enable stable row ids. That's set when the dataset is created. So we should just call this `stable_row_ids`.
```suggestion
    /// Whether the dataset uses stable row ids.
    stable_row_ids: bool,
```